### PR TITLE
add libsquish/1.15

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **libsquish/1.15**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
I don't know if sse2_intrinsics and altivec_intrinsics are good options names so that consumer would understand that if enabled, libsquish uses its own custom vectorization instructions, while if disabled we let compiler generates simd instructions as best it can (which might provide better performance, I don't know).

**Comment by SpaceIm** [comment]
I've removed WINDOWS_EXPORT_ALL_SYMBOLS and replaced by generate_export_header, since:
- there is not a lot of public functions, so it reduces shared lib size.
- libsquish's development branch has also a commit using generate_export_header.

File: recipes/libsquish/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_SQUISH_WITH_OPENMP"] = self.options.openmp
+        if self.settings.arch in self._sse2_compliant_archs:
+            self._cmake.definitions["BUILD_SQUISH_WITH_SSE2"] = self.options.sse2_intrinsics
```
Comment (uilianries) [inline recipes/libsquish/all/conanfile.py]: Also, you could use `self.options.get_safe("sse2_intrinsics") or False`

File: recipes/libsquish/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def package(self):
+        self.copy("LICENSE.txt", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
```
Comment (uilianries) [inline recipes/libsquish/all/conanfile.py]:
```suggestion
        self.cpp_info.libs = tools.collect_libs(self)
        if self.settings.os == "Linux":
            self.cpp_info.system_libs = ["m"]
```

Comment (SpaceIm) [reply recipes/libsquish/all/conanfile.py]: Is it really required since libsquish include cmath, not math.h?

Comment (uilianries) [reply recipes/libsquish/all/conanfile.py]: yes, checked using ldd.

Comment (SpaceIm) [reply recipes/libsquish/all/conanfile.py]: ok, fixed

**Pull request merged**